core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/services/impl/OptionProviderServiceImpl.java

```diff
@@ -120,19 +151,66 @@ private List<Option> getOptions(Resource dataSource,
                 : CoreConstants.PARAMETER_ID;
         return StreamSupport.stream(dataSource.getChildren().spliterator(), false)
                 .filter(child -> !child.getName().equals(JcrConstants.JCR_CONTENT)) // jcr:content nodes are excluded
-                .map(child -> Option.builder()                                      // from the option sources
-                        .resource(child)
-                        .textMember(parameters.getTextMember())
-                        .valueMember(StringUtils.defaultString(parameters.getValueMember(), defaultValueMember))
-                        .attributeMembers(parameters.getAttributeMembers())
-                        .attributes(parameters.getAttributes())
-                        .textTransform(parameters.getTextTransform())
-                        .valueTransform(parameters.getValueTransform())
-                        .build())
+                .map(child -> getOption(child, parameters, defaultValueMember))
                 .filter(Option::isValid)
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Called from {@link OptionProviderServiceImpl#getOptions(SlingHttpServletRequest)} to extract a list
+     * of {@link Option} items from the particular json
+     * @param jsonNode {@code JsonNode} instance representing json from GET request
+     * @param parameters Path-related user settings that came with the request
+     * @return {@code List<DataSourceOption>} object, or an empty list
+     */
+    private List<Option> getOptions(ResourceResolver resourceResolver, PathParameters parameters, JsonNode jsonNode) {
+        if (jsonNode.isArray()) {
+            return StreamSupport.stream(Spliterators.spliteratorUnknownSize(jsonNode.elements(), Spliterator.ORDERED), false)
+                .map(element -> {
+                    Map<String, Object> map = new HashMap<>();
+                    map.put(parameters.getTextMember(), element.get(parameters.getTextMember()).textValue());
+                    map.put(parameters.getValueMember(), element.get(parameters.getValueMember()).textValue());
+                    return map;
+                })
```

Comment:
you can use some another map implementation or instantiation method to be able to avoid lambda with brackets.

Look at the com.google.common.collect.ImmutableMap :

`ImmutableMap.of("key1", "value1", "key2", "value2");`

or if you know what madness it:

`Map<String, String> map = Stream.of(new String[][] {
  { "Hello", "World" }, 
  { "John", "Doe" }, 
}).collect(Collectors.toMap(data -> data[0], data -> data[1]));`


`Map<String, Integer> map = Stream.of(
  new AbstractMap.SimpleImmutableEntry<>(key1, value1),    
  new AbstractMap.SimpleImmutableEntry<>(key2, value2))
  .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));`

Reply:
![image](https://user-images.githubusercontent.com/70597105/179925968-ba417ec1-6861-4992-8ef1-beacdaaae8da.png)
I tried, but it doesn't work(

Reply:
Ugh, Guava just seems to be missing from the core POM. I think it can be added with no consequence, it is anyway mentioned in the main POM file.
We add dependencies strictly in the alphabetical order of groupId-s, then artifactId-s, except for the uber-jar